src/main/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferService.java

```diff
@@ -77,6 +77,10 @@ public void updateConversationTransferStatus(UUID inboundConversationId, Convers
 
     public void updateConversationTransferStatusWithFailure(UUID inboundConversationId, String failureCode) {
         transferRepository.updateConversationStatusWithFailure(inboundConversationId, failureCode);
+
+        // remove the in-memory activity to end the conversation
```

Comment:
If `Timestamp` is removed from the below method call, this comment is unnessesary due to it being readable as-is:

- updateConversationTransferStatusWithFailure
- activityService.removeConversationActivity(inboundConversationId)